db/migrate/20201022204510_add_extra_races_to_patient.rb

```diff
@@ -0,0 +1,9 @@
+class AddExtraRacesToPatient < ActiveRecord::Migration[6.0]
+  def change
+    add_column :patients, :unknown, :boolean
+    add_column :patients, :other, :boolean
+    add_column :patients, :refused_to_answer, :boolean
+
+    add_column :patients, :races, :string, array: true, default: [].to_yaml
```

Comment:
What is your thinking for storing these like this versus keeping them as fields on the record?

Reply:
Keeping them as separate fields worked fine, but it has a few problems after extending: adding names like "unknown" and "other" to patients can be confusing, and it can be hard to tell at a glance what races a patient object might have looking at the raw data. The array structure avoids the weirdly named booleans, puts all race data in a single variable so you can at a glance see all the relevant data (though that's negated some by the fact I had to use the FHIR codes instead of proper names later on), and it's much more extensible: If we ever decide to allow for CDC race categories, of which there are over 900, this refactor will make the change much easier.

Reply:
Keeping them as separate fields would provide benefit in the event that we need to query by race. As it is currently written if we need to find all patients by a particular race it gets expensive because each field of race needs to get parsed as YAML, converted to a Ruby object, and then filtered in Ruby. By leaving race as individual attributes the database can do the filtering more quickly and we avoid potentially expensive YAML parsing over lots of patients.

Reply:
I think it's safe to say we won't need to support more than the OMB codes for race.

Reply:
I'm actually not sure about that: we already support "other", and I foresee a distinct likelihood that someone will wind up wanting that to become "other (please specify)".  

It's a good point about querying: I think there's probably a way to avoid filtering in Ruby using the FIND_IN_SET sql function, but it merits further testing. I can try to stress-test it today by upping the number of patients in the demo rake task.

Reply:
What is the activerecord support for find in set?

Reply:
It's a bit hacky and could be better, but the query:

SELECT first_name, races FROM patients WHERE races LIKE '%2106-3%';

gets the first name and full race array for all white patients in the database without ruby parsing, and completes in 0.0 seconds when run on my local machine in a database with 3907 patients. That performance should scale at a similar rate to any other query, since the race array can only have at most 5 entries.

Reply:
I really want to err on the side of simple for this. This strikes me as a real good case of YAGNI. Keep the OMB fields as individual columns, and add the addition null flavors as alternatives. All paths for updating these (API/UI/Import) should enforce exclusivity.

Reply:
@holmesie sounds good. I've committed my final draft of the array version to maintain a record of it in the repo, and will move it back to parameters in the next commit.